sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LocalRelation.scala

```diff
@@ -45,6 +45,9 @@ object LocalRelation {
 case class LocalRelation(output: Seq[Attribute], data: Seq[InternalRow] = Nil)
   extends LeafNode with analysis.MultiInstanceRelation {
 
+  // A local relation must have resolved output.
+  require(output.forall(_.resolved), "Unresolved attributes found when constructing LocalRelation.")
```

Comment:
The reported error was `cannot resolve 'i' given input columns: []` which is pretty confusing since every node in the plan does actually have `output`.  The reason for this error was we were constructing a `LocalRelation` that itself was unresolved.  Since this isn't a valid plan, I added this check to force an earlier failure.